File: android/src/main/kotlin/dev/masel/synced_bloc/SyncSlave.kt
package dev.masel.synced_bloc

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

internal class SyncSlave(
    binaryMessenger: BinaryMessenger,
    val masterId: String,
    val slaveId: String
) {
    private val channel = MethodChannel(binaryMessenger, "dev.masel.synced_bloc.slave.$masterId.$slaveId")

    fun notifyMasterChange() =
        channel.invokeMethod("notifyMasterChange", null)
}

File: ios/Classes/SyncSlave.swift
import Flutter

internal struct SyncSlave: Equatable {
    private let channel: FlutterMethodChannel
    let masterId: String
    let slaveId: String

    init(binaryMessenger: FlutterBinaryMessenger, masterId: String, slaveId: String) {
        self.channel = FlutterMethodChannel(name: "dev.masel.synced_bloc.slave.\(masterId).\(slaveId)", binaryMessenger: binaryMessenger)
        self.masterId = masterId
        self.slaveId = slaveId
    }

    func notifyMasterChange() {
        channel.invokeMethod("notifyMasterChange", arguments: nil)
    }

    static func == (lhs: SyncSlave, rhs: SyncSlave) -> Bool {
        return lhs.masterId == rhs.masterId && lhs.slaveId == rhs.slaveId
    }
}
